compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -443,6 +443,9 @@ pub struct Body<'tcx> {
     /// If `-Cinstrument-coverage` is not active, or if an individual function
     /// is not eligible for coverage, then this should always be `None`.
     pub function_coverage_info: Option<Box<coverage::FunctionCoverageInfo>>,
+
+    /// Whether optimization is disabled by `#[optimize(none)]`
+    pub optimization_disabled: bool,
```

Comment:
Do we need to store this in the body, or can we get away with querying it on demand?